superset-frontend/src/components/Select/Select.test.tsx

```diff
@@ -169,34 +180,65 @@ test('sort the options using a custom sort comparator', async () => {
   });
 });
 
-test('displays the selected values first', async () => {
-  render(<Select {...defaultProps} mode="multiple" />);
-  const option3 = OPTIONS[2].label;
-  const option8 = OPTIONS[7].label;
+// eslint-disable-next-line jest/expect-expect
+test('should sort selected to top when in single mode', async () => {
+  render(<Select {...defaultProps} mode="single" />);
+  const originalLabels = OPTIONS.map(option => option.label);
   await open();
-  userEvent.click(await findSelectOption(option3));
-  userEvent.click(await findSelectOption(option8));
+  userEvent.click(await findSelectOption(originalLabels[1]));
+  // after selection, keep the original order
+  await checkOrder(originalLabels);
+
+  // order selected to top when reopen
   await type('{esc}');
   await open();
-  const sortedOptions = await findAllSelectOptions();
-  expect(sortedOptions[0]).toHaveTextContent(option3);
-  expect(sortedOptions[1]).toHaveTextContent(option8);
+  let labels = originalLabels.slice();
+  labels = labels.splice(1, 1).concat(labels);
+  await checkOrder(labels);
+
+  // keep clicking other items, the updated order should still based on
+  // original order
+  userEvent.click(await findSelectOption(originalLabels[5]));
+  await checkOrder(labels);
+  await type('{esc}');
+  await open();
+  labels = originalLabels.slice();
+  labels = labels.splice(5, 1).concat(labels);
+  await checkOrder(labels);
+
+  // should revert to original order
+  clearAll();
+  await type('{esc}');
+  await open();
+  await checkOrder(originalLabels);
 });
 
-test('displays the original order when unselecting', async () => {
+// eslint-disable-next-line jest/expect-expect
```

Comment:
May I suggest renaming `checkOrder` to `matchOrder`, and returning a `boolean`? This way we could have the `expect` clauses inside the tests and avoid disabling lint rules.

Reply:
good suggestion!